lib/internal/modules/cjs/loader.js

```diff
@@ -121,15 +121,48 @@ Module._extensions = Object.create(null);
 var modulePaths = [];
 Module.globalPaths = [];
 
-Module.wrap = function(script) {
+let patched = false;
+
+// eslint-disable-next-line func-style
+let wrap = function(script) {
   return Module.wrapper[0] + script + Module.wrapper[1];
 };
 
-Module.wrapper = [
+const wrapper = [
   '(function (exports, require, module, __filename, __dirname) { ',
   '\n});'
 ];
 
+let wrapperProxy = new Proxy(wrapper, {
+  set(obj, prop, value) {
+    if (prop === '0' || prop === '1') {
```

Comment:
```js
set(target, property, value, receiver) {
  patched = true;
  return Reflect.set(target, property, value, receiver);
}
```

Reply:
we only need it on defineProperty, set will call defineProperty for assignment ops